internal/ecliptic.js

```diff
@@ -1,238 +1,113 @@
 
-module.exports.owner = (contracts) => {
-  return contracts.ecliptic.methods.owner().call();
-}
+module.exports.owner = (contracts) =>
+  contracts.ecliptic.methods.owner().call()
 
-module.exports.balanceOf = (contracts, address) => {
-  return contracts.ecliptic.methods.balanceOf(address).call();
-}
+module.exports.balanceOf = (contracts, address) =>
+  contracts.ecliptic.methods.balanceOf(address).call()
 
-module.exports.ownerOf = (contracts, point) => {
-  return contracts.ecliptic.methods.ownerOf(point).call();
-}
+module.exports.ownerOf = (contracts, point) =>
+  contracts.ecliptic.methods.ownerOf(point).call()
 
-module.exports.exists = (contracts, point) => {
-  return contracts.ecliptic.methods.exists(point).call();
-}
+module.exports.exists = (contracts, point) =>
+  contracts.ecliptic.methods.exists(point).call()
 
-module.exports.getApproved = (contracts, point) => {
-  return contracts.ecliptic.methods.getApproved(point).call();
-}
+module.exports.getApproved = (contracts, point) =>
+  contracts.ecliptic.methods.getApproved(point).call()
 
-module.exports.isApprovedForAll = (contracts, owner, operator) => {
-  return contracts.ecliptic.methods.isApprovedForAll(owner, operator).call();
-}
+module.exports.isApprovedForAll = (contracts, owner, operator) =>
+  contracts.ecliptic.methods.isApprovedForAll(owner, operator).call()
 
-module.exports.getSpawnLimit = (contracts, point, time) => {
-  return contracts.ecliptic.methods.getSpawnLimit(point, time).call();
-}
+module.exports.getSpawnLimit = (contracts, point, time) =>
+  contracts.ecliptic.methods.getSpawnLimit(point, time).call()
 
-module.exports.canEscapeTo = (contracts, point, sponsor) => {
-  return contracts.ecliptic.methods.canEscapeTo(point, sponsor).call();
-}
+module.exports.canEscapeTo = (contracts, point, sponsor) =>
+  contracts.ecliptic.methods.canEscapeTo(point, sponsor).call()
 
-// NB (jtobin):
-//
-//   The following is extremely repetitive.  Could obviously abstract via
-//   function pointers/variadic arguments, e.g. something like:
-//
-//     function txFromMethod(method, ...args) {
-//       let addr = contracts.ecliptic._address;
-//       let data = contracts.ecliptic.methods.method.apply(this, args);
-//       let abi  = data.encodeABI();
-//       return tx(addr, abi, 0);
-//     }
-//
-//   Except handling 'method' appropriately.  But, given there are only a
-//   handful of functions, it's probably not worth doing any metaprogramming.
-
-const tx = (to, data, value) => ({
-  to: to,
-  data: data,
-  value: value || 0x0
-});
-
-module.exports.safeTransferFrom = (contracts, _from, _to, _point) => {
-  let addr = contracts.ecliptic._address;
-  let data = contracts.ecliptic.methods.safeTransferFrom(
-               _from, _to, _point);
-  let abi  = data.encodeABI();
-  return tx(addr, abi, 0);
+function txFromMethod(contracts, method, ...args) {
```

Comment:
Why not put this to use in the other method files also? You could probably throw this into a helper lib, constructing this function by passing in a `contracts.whatever` object.

Reply:
« Bonne idée » !